lib/pbench/agent/tool_group.py

```diff
@@ -173,3 +174,14 @@ def archive(self, target_dir: Path):
             0
         """
         shutil.copytree(str(self.tg_dir), target_dir / self.tg_dir.name, symlinks=False)
+
+
+def gen_tool_groups(pbench_run: str) -> Iterable[ToolGroup]:
+    """Generate a series of ToolGroup objects for each on-disk tool group
+    found in the given pbench run directory.
+    """
+    for tg_dir in Path(pbench_run).glob(f"{ToolGroup.TOOL_GROUP_PREFIX}-*"):
+        # All on-disk tool group directories will have names that look like
+        # above, where the prefix also contains a '-'.  To strip the prefix,
+        # we split the string and use the last component of the split.
+        yield ToolGroup(tg_dir.name.split("-", 2)[2], pbench_run)
```

Comment:
If what we want is the "last component", then maybe the index should be `-1` instead of `2`.

However, it would be better to just remove the prefix (e.g., via [`str.removeprefix()`](https://docs.python.org/3/library/stdtypes.html#str.removeprefix)...unfortunately, that's new in 3.9, but you could use something like `string[len(ToolGroup.TOOL_GROUP_PREFIX) + 1:]`).

And, I would reword the comment to read something like
> Extract the tool group name from the directory name by removing the group prefix.